rest-servlet/src/main/java/org/hawkular/inventory/rest/interceptors/LoggingInterceptor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.rest.interceptors;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.Provider;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.io.IOUtils;
+import org.hawkular.inventory.rest.RestApiLogger;
+import org.hawkular.inventory.rest.json.JacksonConfig;
+
+import org.jboss.resteasy.annotations.interception.ServerInterceptor;
+
+/**
+ * @author Jirka Kremser
+ * @since 0.2.0
+ */
+
+@Provider
+@ServerInterceptor
+public class LoggingInterceptor implements ContainerRequestFilter {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    static {
+        JacksonConfig.initializeObjectMapper(MAPPER);
+    }
+
+    @Override
+    public void filter(ContainerRequestContext containerRequestContext) throws IOException {
+        // perhaps better than logger lvl could be a system property passed by -Dx=y
```

Comment:
You could also reserve a special logger category for this so that users can switch this on/off independently of debug logging of the rest of the REST api - should be as easy as adding a new constant to the `RestApiLogger` interface and using it here.